### PR TITLE
Add run notebook in dedicated extension host command

### PR DESCRIPTION
**Pull request by rebornix**
Fixes https://github.com/microsoft/vscode/issues/140374

-   [ ] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [ ] Title summarizes what is changing.
-   [ ] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [ ] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

File: src/client/common/serviceRegistry.ts [hunk 1]
```diff
@@ -88,6 +89,10 @@ export function registerTypes(serviceManager: IServiceManager) {
         IExtensionSingleActivationService,
         ReloadVSCodeCommandHandler
     );
+    serviceManager.addSingleton<IExtensionSingleActivationService>(
```
[the remark this answers is not in the document]
Comment (rebornix) [reply src/client/common/serviceRegistry.ts:92]: oh I see, let me rebase

**Review by rchiodo — CHANGES_REQUESTED** [review]
Package-lock.json shouldn't be updated. Ian's going to do that after the release, but it looks like your running of npm install updated it.

File: src/platform/common/application/commands/runInDedicatedExtensionHost.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+import { inject, injectable } from 'inversify';
+import { ConfigurationTarget, extensions } from 'vscode';
+import { IExtensionSingleActivationService } from '../../../activation/types';
+import { PylanceExtension, PythonExtension } from '../../../datascience/constants';
+import { noop } from '../../utils/misc';
+import { ICommandManager, IWorkspaceService } from '../types';
+
+/**
+ * Prompts user to reload VS Code with a custom message, and reloads if necessary.
```
Comment (amunger) [inline src/platform/common/application/commands/runInDedicatedExtensionHost.ts:14]: Does it actually prompt? I don't see that happening in the code

**Review by DonJayamanne — CHANGES_REQUESTED** [review]
not sure why the package-lock is changing.
we have a debt item to address this (@rchiodo @IanMatthewHuff  Ajay aware of this issue) please could you revert the changes to the package lock file

**Review by rchiodo — APPROVED** [review]

**Comment by IanMatthewHuff** [comment]
Updated the package lock. Something is off with linting still. Looking into it quick.

**Comment by IanMatthewHuff** [comment]
Small tslint fix on the affinity max check.

**Review by DonJayamanne — APPROVED** [review]

**Pull request merged**